internal/session/repository_connection.go

```diff
@@ -184,6 +184,71 @@ func (r *ConnectionRepository) ListConnectionsBySessionId(ctx context.Context, s
 	return connections, nil
 }
 
+// CloseConnectionResp is just a wrapper for the response from CloseConnections.
+// It wraps the connection and its states for each connection closed.
+type CloseConnectionResp struct {
+	Connection       *Connection
+	ConnectionStates []*ConnectionState
+}
+
+// CloseConnections set's a connection's state to "closed" in the repo.  It's
+// called by a worker after it's closed a connection between the client and the
+// endpoint
+func (r *ConnectionRepository) CloseConnections(ctx context.Context, closeWith []CloseWith, _ ...Option) ([]CloseConnectionResp, error) {
```

Comment:
I think this can be unexported now since it should only be called by the domain service.